### PR TITLE
add comments; disable old NewConversationRemote in mock

### PR DESCRIPTION
**Pull request by songgao**
Mostly just some notes in comments that goes with https://github.com/keybase/gregor/pull/183

### CODE REVIEW

**Comment by songgao** [comment]
Decided to do https://github.com/keybase/gregor/pull/183 her in client. Please take another look? @patrickxb @mmaxim Thanks!

File: go/service/chat_local_test.go [hunk 1]
```diff
@@ -65,6 +65,28 @@ func TestNewConversationLocal(t *testing.T) {
...
+	ctc := makeChatTestContext(t, "NewConversationLocal")
+	defer ctc.tc.Cleanup()
+
+	mustCreateConversationForTest(t, ctc, chat1.TopicType_CHAT, "t_alice")
+
+	if _, err := ctc.h.NewConversationLocal(context.Background(), chat1.ConversationInfoLocal{
+		TlfName:   "t_alice," + ctc.world.me.Username,
+		TopicType: chat1.TopicType_CHAT,
+	}); err == nil {
+		t.Fatalf("creating CHAT conversation twice did not fail")
```
Comment (patrickxb) [inline go/service/chat_local_test.go]: Wasn't the point that NewConversation would return an existing conversation, not an error?

Comment (mmaxim) [inline go/service/chat_local_test.go]: Yeah I made a similar comment on the Gregor PR.

Comment (songgao) [inline go/service/chat_local_test.go]: Yes. Will update!

File: go/service/chat_local.go [hunk 2]
```diff
@@ -96,25 +97,51 @@ func (h *chatLocalHandler) NewConversationLocal(ctx context.Context, info chat1.
...
+		// is because we enforce one CHAT conversation per TLF for now.
+
 		res, err := h.remoteClient().NewConversationRemote2(ctx, chat1.NewConversationRemote2Arg{
 			IdTriple:   triple,
 			TLFMessage: *firstMessageBoxed,
 		})
 		if err != nil {
-			return err
+			if cerr, ok := err.(libkb.ChatConvExistsError); ok {
+				if triple.TopicType == chat1.TopicType_CHAT /* TODO: check for error type */ {
```
Comment (mmaxim) [inline go/service/chat_local.go]: I this TODO still relevant?

File: go/service/chat_local.go [hunk 3]
```diff
@@ -96,25 +97,51 @@ func (h *chatLocalHandler) NewConversationLocal(ctx context.Context, info chat1.
 	}
 	info.TlfName = string(res.CanonicalName)
 
-	if err = retryWithoutBackoffUpToNTimesUntilNoError(3, func() (err error) {
-		if triple.TopicID, err = libkb.NewChatTopicID(); err != nil {
-			return fmt.Errorf("error creating topic ID: %s", err)
+	if err = retryWithoutBackoffUpToNTimesUntilNoError(3, func() (retriable bool, err error) {
```
Comment (mmaxim) [inline go/service/chat_local.go]: Maybe just make this a `for` loop?

**Comment by mmaxim** [comment]
Couple small questions, but otherwise looks 👍

File: go/service/chat_local.go (hunk 2 above)
Comment (songgao) [inline go/service/chat_local.go]: Nope I'll remote it before merging. Thanks Mike!

File: go/service/chat_local.go (hunk 3 above)
Comment (songgao) [inline go/service/chat_local.go]: Sounds good!

**Pull request merged**